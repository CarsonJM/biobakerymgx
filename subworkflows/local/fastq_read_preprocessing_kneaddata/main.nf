include { KNEADDATA_DATABASE            } from '../../modules/local/kneaddata/database/main'
include { KNEADDATA_KNEADDATA           } from '../../modules/local/kneaddata/kneaddata/main'
include { KNEADDATA_READCOUNTS          } from '../../modules/local/kneaddata/readcounts/main'

workflow FASTQ_READ_PREPROCESSING_KNEADDATA {

    take:
    ch_raw_reads_fastq_gz   // channel: [ val(meta), [ reads_1.fastq.gz, reads_2.fastq.gz ] ]

    main:

    ch_versions = Channel.empty()

    //
    // MODULE: Download KneadData database
    //
    ch_kneaddata_db = KNEADDATA_DATABASE ().kneaddata_db
    ch_versions = ch_versions.mix(KNEADDATA_DATABASE.out.versions)

    //
    // MODULE: Trim and remove human reads
    //
    ch_preprocessed_reads_fastq_gz = KNEADDATA_KNEADDATA ( ch_raw_short_reads, ch_kneaddata_db ).preprocessed_reads
    ch_versions = ch_versions.mix(KNEADDATA_KNEADDATA.out.versions)

    emit:
    reads = KNEADDATA_KNEADDATA.out.reads // channel: [ val(meta), [ reads ] ]

    versions = ch_versions // channel: [ versions.yml ]
}
