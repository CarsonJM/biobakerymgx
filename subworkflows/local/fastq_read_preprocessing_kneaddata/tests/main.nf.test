nextflow_workflow {

    name "Test Subworkflow: FASTQ_READ_PREPROCESSING_KNEADDATA"
    script "../main.nf"
    workflow "FASTQ_READ_PREPROCESSING_KNEADDATA"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "fastq_read_preprocessing_kneaddata"
    tag "fastq_read_preprocessing_kneaddata_default"


    test(" ['sarscov2']['illumina']['test_fastq_gz'] ") {

        when {
            params {
                outdir      = "$outputDir"
                max_cpus = 1
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                    ],
                    [
                        [ id: 'test2' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test2_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test2_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test(" ['sarscov2']['illumina']['test_fastq_gz'] - stub ") {

        options "-stub"

        when {
            params {
                outdir      = "$outputDir"
                max_cpus    = 1
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                    ],
                    [
                        [ id: 'test2' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test2_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test2_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
