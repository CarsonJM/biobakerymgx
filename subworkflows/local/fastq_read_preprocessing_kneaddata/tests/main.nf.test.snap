{
    " fasta.gz + 'human_genome' ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test2_kneaddata_paired_2.fastq.gz:md5,7c8a08782ad6705eced33d11d5032859"
                        ]
                    ],
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test_kneaddata_paired_2.fastq.gz:md5,7c8a08782ad6705eced33d11d5032859"
                        ]
                    ]
                ],
                "1": [
                    "all_samples_kneaddata_read_counts.tsv:md5,0cbc1c285d40c25876ce0675fe36f34b"
                ],
                "2": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8",
                    "versions.yml:md5,c97c9af1b0da5c6c85c297ad365653cc"
                ],
                "preprocessed_fastq_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test2_kneaddata_paired_2.fastq.gz:md5,7c8a08782ad6705eced33d11d5032859"
                        ]
                    ],
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test_kneaddata_paired_2.fastq.gz:md5,7c8a08782ad6705eced33d11d5032859"
                        ]
                    ]
                ],
                "read_counts_tsv": [
                    "all_samples_kneaddata_read_counts.tsv:md5,0cbc1c285d40c25876ce0675fe36f34b"
                ],
                "versions": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8",
                    "versions.yml:md5,c97c9af1b0da5c6c85c297ad365653cc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-25T21:47:34.07117763"
    }
}